book_scraper/spiders/books.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+import scrapy
+
+RATING_MAPPER = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/catalogue/page-1.html"]
+
+    def parse(self, response, **kwargs):
+        total_pages = response.css(
+            "li.current::text"
+        ).re_first(r"Page \d+ of (\d+)")
+        if total_pages:
+            total_pages = int(total_pages)
+        else:
+            total_pages = 1
+        for page in range(1, total_pages + 1):
+            url = f"https://books.toscrape.com/catalogue/page-{page}.html"
+            yield scrapy.Request(url=url, callback=self.parse_page)
+
+    def parse_page(self, response):
+        for book_url in response.css(
+                ".image_container > a::attr(" "href)").getall():
+            url = "https://books.toscrape.com/catalogue/" + book_url

```

Comment:
The URL construction assumes that `book_url` does not start with a slash. If it does, this could lead to incorrect URLs. Consider using `urljoin` from `scrapy.utils.response` to handle relative URLs properly.